### DISCUSSION TITLE
Docstrings for `#[new]` method implementations

### DISCUSSION THREAD

**Discussion by djc** [PyO3/pyo3#4280] [Questions] [answered]
When I write a docstring on a `#[new] fn new()` method in a `#[pymethods]` `impl` block, I would expect this to show up in the `help()` output for the type. Instead, the `help()` output seems to end up with a default docstring:

```
 |  Static methods defined here:
 |
 |  __new__(*args, **kwargs)
 |      Create and return a new object.  See help(type) for accurate signature.
```

Is there no way in Python's data model to enable a specific help string for this? The docstring for the type ends up making it into the output sort of next to the constructor type signature, so I guess I could use that. Just seems surprising.

**Comment by davidhewitt** [comment, marked as answer]
Sorry for the slow reply. I think this is just an oversight that we should be able to improve, I opened #4326 to track this.

**Reply by djc** [reply]
Thanks!
